torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -183,11 +186,19 @@ auto combine_self_args(PyObject *self, PyObject *args) -> py::tuple {
   return args_;
 }
 
+// TODO: I'm not sure if I should call this __torch_function__ or
+// torch_function.  The former makes it easier to take an existing
+// Tensor-like __torch_function__ object and turn it into a mode;
+// but in general modes don't have to be Tensor-like (and we will
+// improperly accept mode objects as arguments when they shouldn't
```

Comment:
Yes, if we change the name then we need a special case for "pass Tensor subclass as a mode" so that we look at the old name. Score another point for the "reuse the same name" team